### DISCUSSION TITLE
Dynamic handles in partials

### DISCUSSION THREAD

**Discussion by chopperoon** [statamic/cms#5141] [Q&A]
I'm using multiple instances of a fieldset within a template and each fieldset has its own prefix. I'm trying to use a partial to show the fieldset each time and thought that I could pass the prefix through as a variable and then use it in the partial to form a field handle name, like this:

`{{ partial:page-claim prefix="my_prefix" }}`

And then use something like `{{ {my_prefix}my_field_handle }}` within the partial but this doesn't work. Is it possible to achieve what I'm trying to do?

**Comment by edalzell** [comment]
What do you need to do with `{{ {my_prefix}my_field_handle }}`? Like how do you want to use it in the partial?

**Comment by jackmcdade** [comment]
@JohnathonKoster how would you handle this in 3.3?

**Reply by JohnathonKoster** [reply]
@jackmcdade With dynamic variable references (prefixed with `@`):

Partial:
```
{{
    dynamicVar = prefix + 'my_field_handle'
}}

Field Name: {{ dynamicVar }}<br />
Value: {{ @dynamicVar }}
```

Other Antlers:
```
{{ partial:page-claim prefix="my_prefix" }}
```

Produces the output:

```
Dynamic Variable Test

Field Name: my_prefixmy_field_handle
Value: Test Value!
```

I keep going back and forth on whether to make them work on strings directly, and avoid that intermediate var. This would look something like the following to construct and evaluate dynamic variables:

```
{{ @'{prefix}my_field_handle' }}
```

![dynamic_vars](https://user-images.githubusercontent.com/5232890/152417681-2a1a058d-4cff-432e-a5b1-821a6cbd04b9.png)

**Comment by JohnathonKoster** [comment]
What a late follow up on this!

With the Runtime version, the following is now possible:

```antlers
{{ partial:page-claim handle_prefix="my_prefix_" }}
```

Then, within the partial you can continue referencing your variables like normal:

```antlers
{{ my_field_handle }}
```

When the `handle_prefix` is present, and set to `my_prefix_`, if a variable with the handle `my_prefx_my_field_handle` exists, that will take precedence 👍

I wrote up a few more examples here: https://stillat.com/blog/2022/04/22/managing-fieldset-handles-with-antlers-runtime
